core/src/main/scala/sttp/tapir/server/interpreter/RequestBody.scala

```diff
@@ -1,10 +1,20 @@
 package sttp.tapir.server.interpreter
 
 import sttp.capabilities.Streams
-import sttp.tapir.RawBodyType
+import sttp.model.Part
+import sttp.tapir.{RawBodyType, RawPart}
+
+import java.io.File
 
 trait RequestBody[F[_], S] {
   val streams: Streams[S]
-  def toRaw[R](bodyType: RawBodyType[R]): F[R]
+  def toRaw[R](bodyType: RawBodyType[R]): F[RawValue[R]]
   def toStream(): streams.BinaryStream
 }
+
+case class RawValue[R](value: R, tmpFiles: Seq[File] = Nil)
```

Comment:
[minor] maybe: `createdFiles`. They don't really have to be temporary, that's just the default implementation